arcgis-ios-sdk-samples/Maps/Apply scheduled updates to preplanned map area/ApplyScheduledUpdatesToPreplannedMapAreaViewController.swift

```diff
@@ -0,0 +1,190 @@
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import ArcGIS
+
+/// A view controller that manages the interface of the Apply Scheduled Updates
+/// to Preplanned Map Area sample.
+class ApplyScheduledUpdatesToPreplannedMapAreaViewController: UIViewController {
+    /// The map view managed by the view controller.
+    @IBOutlet weak var mapView: AGSMapView!
+    
+    /// The mobile map package used by this sample.
+    let mobileMapPackage: AGSMobileMapPackage!
+    /// The sync task used to check for scheduled updates.
+    var offlineMapSyncTask: AGSOfflineMapSyncTask!
+    /// The sync job used to apply updates to the offline map.
+    var offlineMapSyncJob: AGSOfflineMapSyncJob!
+    
+    required init?(coder: NSCoder) {
+        let mobileMapPackageURL = Bundle.main.url(forResource: "canyonlands", withExtension: nil)!
+        do {
+            let temporaryDirectoryURL = try FileManager.default.url(
+                for: .itemReplacementDirectory,
+                in: .userDomainMask,
+                appropriateFor: mobileMapPackageURL,
+                create: true
+            )
+            let temporaryMobileMapPackageURL = temporaryDirectoryURL.appendingPathComponent(ProcessInfo.processInfo.globallyUniqueString)
+            try FileManager.default.copyItem(at: mobileMapPackageURL, to: temporaryMobileMapPackageURL)
+            mobileMapPackage = AGSMobileMapPackage(fileURL: temporaryMobileMapPackageURL)
+        } catch {
+            print("Error setting up mobile map package: \(error)")
+            mobileMapPackage = nil
+        }
+        
+        super.init(coder: coder)
+        
+        mobileMapPackage?.load { [weak self] (error) in
+            let result: Result<Void, Error>
+            if let error = error {
+                result = .failure(error)
+            } else {
+                result = .success(())
+            }
+            self?.mobileMapPackageDidLoad(with: result)
+        }
+    }
+    
+    deinit {
+        if let mobileMapPackage = mobileMapPackage {
+            mobileMapPackage.close()
+            try? FileManager.default.removeItem(at: mobileMapPackage.fileURL)
+        }
+    }
+    
+    /// Called in response to the mobile map package load operation completing.
+    /// - Parameter result: The result of the load operation.
+    func mobileMapPackageDidLoad(with result: Result<Void, Error>) {
+        switch result {
+        case .success:
+            let map = self.mobileMapPackage.maps.first!
+            self.loadViewIfNeeded()
+            self.mapView.map = map
```

Comment:
Do you need to use `self.` in this line and the two previous lines?

Reply:
No. The references to `self` came from these lines originally being inside a closure. I've removed them.